api_gateway/src/main/kotlin/org/thechance/api_gateway/endpoints/utils/Extentions.kt

```diff
@@ -38,14 +39,23 @@ fun WebSocketServerSession.extractLocalizationHeaderFromWebSocket(): Pair<String
     return Pair(language, countryCode)
 }
 
-fun PipelineContext<Unit, ApplicationCall>.extractPermissions() : List<Int>{
+fun PipelineContext<Unit, ApplicationCall>.extractPermission(): Int {
     val tokenClaim = call.principal<JWTPrincipal>()
-    return tokenClaim?.payload?.getClaim("permissions")?.asString()
-        ?.split(",")?.mapNotNull { it.toIntOrNull() } ?: emptyList()
+    return tokenClaim?.payload?.getClaim("permission")?.asString()?.toInt() ?: -1
```

Comment:
first : rename tokenClaim to principle
second : you don't need casting to string just type
 return principal?.getClaim("permission", Int::class) ?: -1
 and it also petter to make the permission string not hard coded like that 
 make it in constants